Dockerfile.openshift

```diff
@@ -1,18 +1,24 @@
 FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11 AS builder
 WORKDIR /metallb
 
+COPY .git/ .git/
+
 ADD . /metallb
 
 WORKDIR /metallb/frr-metrics
 RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o ./frr-metrics
 
 
 WORKDIR /metallb/controller
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o ./controller
+RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o ./controller -ldflags \
```

Comment:
This won't work when building from ART.
We need to check if `SOURCE_GIT_COMMIT` / `BUILD_VERSION` are defined and use them, or calculate otherwise (for cluster-ci images).

See https://github.com/metal3-io/baremetal-operator/pull/606 for inspiration.

Reply:
why it won't work?

Reply:
From art's faqs: "ART builds are performed in dist-git, with content (but not commits) copied from the source repo. Thus at build time if your code is inspecting the local git repo it is getting unrelated commits and tags from the dist-git repo, not the source repo."